src/Model/Item.php

```diff
@@ -102,6 +102,32 @@ public static function delete($condition, $priority = PRIORITY_HIGH)
 		dba::close($items);
 	}
 
+	/**
+	 * @brief Delete an item for an user and notify others about it - if it was ours
+	 *
+	 * @param array $condition The condition for finding the item entries
+	 * @param integer $uid User who wants to delete this item
+	 */
+	public static function deleteForUser($condition, $uid)
+	{
+		if ($uid == 0) {
+			return;
+		}
+
+		$items = dba::select('item', ['id', 'uid'], $condition);
+		while ($item = dba::fetch($items)) {
+			// "Deleting" global items just means hiding them
+			if ($item['uid'] == 0) {
+				dba::update('user-item', ['hidden' => true], ['iid' => $item['id'], 'uid' => $uid], true);
+			} elseif ($item['uid'] == $uid) {
+				self::deleteById($item['id'], PRIORITY_HIGH);
+			} else {
+				logger('Wrong ownership. Not deleting item ' . $item['id']);
```

Comment:
Is this something we will really need for debugging? I feel like this can happen often because `$condition` is free form.

Reply:
This hopefully should never happen. And because of that I want to make sure to test for it. There are several logger texts out there that I'm checking on a regular basis.